src/ShoppingCartKernel.java

```diff
@@ -0,0 +1,62 @@
+import components.map.Map;
+
+/**
+ * Shopping cart kernel component with primary methods.
+ */
+public interface ShoppingCartKernel {
+
+    /*
+     * As a little note, I am planning on expanding on the design and using
```

Comment:
Love it! If you do this, I would recommend abandoning our discipline for a moment and instead adding a 5th file to the repo for your Product class. Specifically, look up "java records" for an example of how this can be accomplished.